monai/transforms/spatial/array.py

```diff
@@ -768,19 +862,52 @@ def __call__(
                 for axis in range(len(spatial_size_)):
                     anti_aliasing_sigma[axis] = anti_aliasing_sigma[axis] * int(factors[axis] > 1)
             anti_aliasing_filter = GaussianSmooth(sigma=anti_aliasing_sigma)
-            img_ = anti_aliasing_filter(img_)
+            img_ = convert_to_tensor(anti_aliasing_filter(img_), track_meta=False)
+
+        img = convert_to_tensor(img, track_meta=get_track_meta())
+        _mode = look_up_option(self.mode if mode is None else mode, InterpolateMode)
+        _align_corners = self.align_corners if align_corners is None else align_corners
 
         resized = torch.nn.functional.interpolate(
-            input=img_.unsqueeze(0),
-            size=spatial_size_,
-            mode=look_up_option(self.mode if mode is None else mode, InterpolateMode).value,
-            align_corners=self.align_corners if align_corners is None else align_corners,
+            input=img_.unsqueeze(0), size=spatial_size_, mode=_mode, align_corners=_align_corners
         )
         out, *_ = convert_to_dst_type(resized.squeeze(0), img)
+        if get_track_meta():
+            self.update_meta(out, original_sp_size, spatial_size_)
+            self.push_transform(
+                out,
+                orig_size=original_sp_size,
+                extra_info={
+                    "mode": _mode,
+                    "align_corners": _align_corners if _align_corners is not None else TraceKeys.NONE,
+                    "new_dim": len(original_sp_size) - input_ndim,  # additional dims appended
```

Comment:
I don't quite understand what's this "new_dim" used for? Seems like a new feature?

Thanks.

Reply:
that's actually an existing feature for the mismatched shapes:
```py
>>> import torch
>>> x = torch.zeros((1, 32, 32))
>>> import monai
>>> monai.transforms.Resize((33, 33, 2))(x).shape
(1, 33, 33, 2)
```